include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,686 @@
+  const UpdateFlags                   update_flags)
+  : internal_fe_face_values(mapping, fe, quadrature, update_flags)
+  , internal_fe_subface_values(mapping, fe, quadrature, update_flags)
+  , internal_fe_face_values_neighbor(mapping, fe, quadrature, update_flags)
+  , internal_fe_subface_values_neighbor(mapping, fe, quadrature, update_flags)
+  , fe_face_values(nullptr)
+  , fe_face_values_neighbor(nullptr)
+{}
+
+
+
+template <int dim, int spacedim>
+FEInterfaceValues<dim, spacedim>::FEInterfaceValues(
+  const FiniteElement<dim, spacedim> &fe,
+  const Quadrature<dim - 1> &         quadrature,
+  const UpdateFlags                   update_flags)
+  : internal_fe_face_values(StaticMappingQ1<dim, spacedim>::mapping,
+                            fe,
+                            quadrature,
+                            update_flags)
+  , internal_fe_subface_values(StaticMappingQ1<dim, spacedim>::mapping,
+                               fe,
+                               quadrature,
+                               update_flags)
+  , internal_fe_face_values_neighbor(StaticMappingQ1<dim, spacedim>::mapping,
+                                     fe,
+                                     quadrature,
+                                     update_flags)
+  , internal_fe_subface_values_neighbor(StaticMappingQ1<dim, spacedim>::mapping,
+                                        fe,
+                                        quadrature,
+                                        update_flags)
+  , fe_face_values(nullptr)
+  , fe_face_values_neighbor(nullptr)
+{}
+
+
+
+template <int dim, int spacedim>
+template <class CellIteratorType>
+void
+FEInterfaceValues<dim, spacedim>::reinit(
+  const CellIteratorType &cell,
+  const unsigned int &    face_no,
+  const unsigned int &    sub_face_no,
+  const CellIteratorType &cell_neighbor,
+  const unsigned int &    face_no_neighbor,
+  const unsigned int &    sub_face_no_neighbor)
+{
+  if (sub_face_no == numbers::invalid_unsigned_int)
+    {
+      internal_fe_face_values.reinit(cell, face_no);
+      fe_face_values = &internal_fe_face_values;
+    }
+  else
+    {
+      internal_fe_subface_values.reinit(cell, face_no, sub_face_no);
+      fe_face_values = &internal_fe_subface_values;
+    }
+  if (sub_face_no_neighbor == numbers::invalid_unsigned_int)
+    {
+      internal_fe_face_values_neighbor.reinit(cell_neighbor, face_no_neighbor);
+      fe_face_values_neighbor = &internal_fe_face_values_neighbor;
+    }
+  else
+    {
+      internal_fe_subface_values_neighbor.reinit(cell_neighbor,
+                                                 face_no_neighbor,
+                                                 sub_face_no_neighbor);
+      fe_face_values_neighbor = &internal_fe_subface_values_neighbor;
+    }
+
+  // Set up dof mapping and remove duplicates (for continuous elements).
+  // Possible optimization: if the FEs are discontinuous, we can just
+  // combine the vectors v and v2.
```

Comment:
I don't see how you can reliably detect this. You might have a `{FE_RaviartThomas,FE_Nedelec,FE_Q,FE_DGQ}` combination -- some parts of it are continuous, others are not. I'd just not worry about it and remove the comment.